docs/data-sources/tls_configuration.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: "fastly"
+page_title: "Fastly: fastly_tls_configuration"
+sidebar_current: "docs-fastly-datasource-tls_configuration"
+description: |-
+Get information on Fastly TLS configuration.
+---
+
+# fastly_tls_configuration
+
+Use this data source to get the ID of a TLS configuration for use with other resources.
+
+~> **Warning:** The data source's filters are applied using an **AND** boolean operator, so depending on the combination
+of filters, they may become mutually exclusive. The exception to this is `id` which must not be specified in combination
+with any of the others.
+
+~> **Note:** If more or less than a single match is returned by the search, Terraform will fail. Ensure that your search is specific enough to return a single key.
+
+## Example Usage
+
+```hcl
+data "fastly_tls_configuration" "example" {
+  default = true
+}
+
+resource "fastly_tls_activation" "example" {
+  configuration_id = data.fastly_tls_configuration.example.id
+  // ...
+}
+```
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- **default** (Boolean) Signifies whether Fastly will use this configuration as a default when creating a new TLS activation.
+- **http_protocols** (Set of String) HTTP protocols available on the TLS configuration.
+- **id** (String) ID of the TLS configuration obtained from the Fastly API or another data source. Conflicts with all the other filters.
+- **name** (String) Custom name of the TLS configuration.
+- **tls_protocols** (Set of String) TLS protocols available on the TLS configuration.
+- **tls_service** (String) Whether the configuration should support the `PLATFORM` or `CUSTOM` TLS service.
+
+### Read-Only
+
+- **created_at** (String) Timestamp (GMT) when the configuration was created.
+- **dns_records** (Set of Object) The available DNS addresses that can be used to enable TLS for a domain. DNS must be configured for a domain for TLS handshakes to succeed. If enabling TLS on an apex domain (e.g. `example.com`) you must create four A records (or four AAAA records for IPv6 support) using the displayed global A record's IP addresses with your DNS provider. For subdomains and wildcard domains (e.g. `www.example.com` or `*.example.com`) you will need to create a relevant CNAME record. (see [below for nested schema](#nestedatt--dns_records))
+- **updated_at** (String) Timestamp (GMT) when the configuration was last updated.
+
+<a id="nestedatt--dns_records"></a>
+### Nested Schema for `dns_records`
```

Comment:
Same issue as @Integralist raised previously (https://github.com/hashicorp/terraform-plugin-docs/issues/28) - no description on nested schema. I've left it autogenerated because I didn't think the descriptions were too critical, it is reasonably self-explanatory what the fields are for, and it seemed easier to maintain by leaving it autogenerated with the rest of the docs.